sdk/go/pulumi/provider.go

```diff
@@ -0,0 +1,792 @@
+	if err != nil {
+		return nil, err
+	}
+
+	inputs, err := plugin.UnmarshalProperties(req.GetNews(), p.unmarshalOptions("inputs"))
+	if err != nil {
+		return nil, err
+	}
+
+	newState, err := p.args.Update(providerCtx, urn, id, state, inputs, req.GetTimeout(), req.GetIgnoreChanges(),
+		req.GetPreview())
+	if err != nil {
+		return nil, err
+	}
+
+	rpcState, err := plugin.MarshalProperties(newState, p.marshalOptions("newState"))
+	if err != nil {
+		return nil, err
+	}
+
+	return &pulumirpc.UpdateResponse{Properties: rpcState}, nil
+}
+
+// Delete tears down an existing resource with the given ID.  If it fails, the resource is assumed
+// to still exist.
+func (p *providerServer) Delete(ctx context.Context, req *pulumirpc.DeleteRequest) (*pbempty.Empty, error) {
+	if p.args.Delete == nil {
+		return &pbempty.Empty{}, nil
+	}
+
+	providerCtx, err := newProviderContext(ctx, p.engineConn)
+	if err != nil {
+		return nil, err
+	}
+
+	urn, id := resource.URN(req.GetUrn()), resource.ID(req.GetId())
+
+	state, err := plugin.UnmarshalProperties(req.GetProperties(), p.unmarshalOptions("state"))
+	if err != nil {
+		return nil, err
+	}
+
+	if err = p.args.Delete(providerCtx, urn, id, state, req.GetTimeout()); err != nil {
+		return nil, err
+	}
+
+	return &pbempty.Empty{}, nil
+}
+
+// Construct creates a new instance of the provided component resource and returns its state.
+func (p *providerServer) Construct(ctx context.Context,
+	req *pulumirpc.ConstructRequest) (*pulumirpc.ConstructResponse, error) {
+
+	if p.args.Construct == nil {
+		return nil, errors.Errorf("unknown resource type %s", req.GetType())
+	}
+
+	// Configure the RunInfo.
+	runInfo := RunInfo{
+		Project:     req.GetProject(),
+		Stack:       req.GetStack(),
+		Config:      req.GetConfig(),
+		Parallel:    int(req.GetParallel()),
+		DryRun:      req.GetDryRun(),
+		MonitorAddr: req.GetMonitorEndpoint(),
+		Mocks:       nil,
+		engineConn:  p.engineConn,
+	}
+	pulumiCtx, err := NewContext(ctx, runInfo)
+	if err != nil {
+		return nil, errors.Wrap(err, "constructing run context")
+	}
+
+	// Deserialize the inputs and apply appropriate dependencies.
+	inputs := &ConstructInputs{inputs: map[string]constructInput{}}
+	inputDependencies := req.GetInputDependencies()
+	deserializedInputs, err := plugin.UnmarshalProperties(
+		req.GetInputs(),
+		plugin.MarshalOptions{KeepSecrets: true, KeepResources: true, KeepUnknowns: req.GetDryRun()},
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "unmarshaling inputs")
+	}
+	for key, input := range deserializedInputs {
+		k := string(key)
+		var deps []Resource
+		if inputDeps, ok := inputDependencies[k]; ok {
+			deps = make([]Resource, len(inputDeps.GetUrns()))
+			for i, depURN := range inputDeps.GetUrns() {
+				deps[i] = newDependencyResource(URN(depURN))
+			}
+		}
+
+		val, secret, err := unmarshalPropertyValue(pulumiCtx, input)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unmarshaling input %s", k)
```

Comment:
Trying to use https://github.com/pulumi/pulumi-component-provider-go-boilerplate.  I'm seeing that my example program just silently doesn't create the component I specified (`const api = new apigateway.RestAPI("api", {...})` is in my code, and runs, but neither it nor any of the child resources are created, and no error is raised).  

When I look at verbose logs, I see this:


> I0326 14:54:57.706385   26028 eventsink.go:59] RegisterResource RPC finished: resource:api[apigateway:index:RestAPI]; err: Error: 2 UNKNOWN: unmarshaling input routes: missing one or more required arguments, resp: undefined

But then after that the registration still seems to complete "correctly" just without doing anything.  

It appears that the error raised on this line isn't causing the resource operation to fail.  I'm unclear if that's an issue on this side of the RPC contract, or within the engine itself.

Reply:
I believe Evan's Node.js error handling fix has addressed the error not failing the update. The actual issue you were running into was around the lack of prompt value support, which is on the way.